pkg/tso/keyspace_group_manager.go

```diff
@@ -705,6 +723,24 @@ func (kgm *KeyspaceGroupManager) deleteKeyspaceGroup(groupID uint32) {
 	}
 }
 
+// resignElectionMembership resigns the election membership of the given keyspace group by
+// deinitializing the allocator manager, but still keeps the keyspace group info.
+func (kgm *KeyspaceGroupManager) resignElectionMembership(group *endpoint.KeyspaceGroup) {
+	log.Info("resign election membership", zap.Uint32("keyspace-group-id", group.ID))
+
+	kgm.Lock()
+	defer kgm.Unlock()
+
+	am := kgm.ams[group.ID]
+	if am != nil {
+		am.close()
```

Comment:
Why not just resign the leadership rather than close the allocator manager?

Reply:
This is just resigning the leadership. The execution in this function means this node doesn't own any replica of this keyspace group and shouldn't take part in the election group. Since this node won't own any replica/member of this keyspace group's election group and moving a keyspace group' replica out of a tso node/pod is rare operation, we'll close AM permanently until it's moved back in the future.

Reply:
Then I suggest using another method name since the term "resign" usually refers to "resign the leadership for now while still participating in the following election" in the PD code. What about `exitElectionMembership`?

Reply:
changed to exitElectionMembership